src/Orleans.Transactions.Testkit.Xunit/TransactionalStateStorageTestRunner.cs

```diff
@@ -7,7 +7,7 @@
 namespace Orleans.Transactions.TestKit.xUnit
 {
     public abstract class TransactionalStateStorageTestRunnerxUnit<TState> : TransactionalStateStorageTestRunner<TState>
-        where TState: class, ITestState, new()
+        where TState: class, IEquatable<TState>, new()
```

Comment:
>IEquatable<TState> [](start = 29, length = 18)

Given the usage of ShouldBeEquivalentTo() IEquatable isn't necessary.  ShouldBeEquivalentTo() will do an object graph walk to do comparison. Implementers can use either IEquatable<> or EquivalencyAssertionOptions<TState> (look at the second parameter of ShouldBeEquivalentTo()) to control object comparison further if necessary.

Reply:
will do